### ISSUE TITLE
Standardize JDBC Connectors: Default SSL behavior, and add Verify SSL Certificate connection option for more flexibility

### ISSUE THREAD

**Issue by noahkawasakigoogle** [airbytehq/airbyte#9871]
## Tell us about the problem you're trying to solve
Non-standard assumptions about SSL being enabled on JDBC connections (adding `ssql=true` to the JDBC string/properties), and non-standard assumptions about verifying server certificates of the database server. There should be more flexibility in the encryption strategies a customer needs to enable connecting securely and successfully. 

## Describe the solution you’d like

Default the SSL parameters to off, and make it opt-in. 

Add a new connection property to JDBC connectors: `verify_ssl_certificate` which would behave the same way `ssl` behaves by adding in the necessary JDBC params `verifyServerCertificate=true/false` into the connection.

```

      "verify_ssl_certificate": {
        "title": "Verify SSL Certificate",
        "description": "Require verification of server certificate for SSL connections.",
        "type": "boolean",
        "default": false,
        "order": 6
      }
```

Lastly, add JDBC tests for this, an extension of: https://github.com/airbytehq/airbyte/issues/9598

## Describe the alternative you’ve considered or used
Some connectors support `jdbc_url_params` and can work around this, others have a lot of custom logic built-in, like the MSSQL connection. However the connectors that do not support this in some cases cannot enable this security check. 
## Additional context
Different connectors make different assumptions about SSL being turned off/on by default. Not a big deal.

Bigger issue is making assumptions about verifying certificate names defaulting to true/false. This actually poses a blocker for special cases where customers dont have a widely accepted CA signed certificate and use self-signed ones, OR if a company has strict security requirements about always verifying hostnames in certificates and such.

## Are you willing to submit a PR?
Yes

**Comment by alafanechere** [comment]
I think working on this topic could solve this kind of issue https://airbytehq-team.slack.com/archives/C01MFR03D5W/p1643301597317600

**Comment by noahkawasakigoogle** [comment]
Yeah for sure! Those scenarios are exactly what I had in mind when writing this up

**Comment by ntkawasaki** [comment]
Another instance: https://airbytehq.slack.com/archives/C01VDDEGL7M/p1645621360020509
